src/validations/javascript/jshint.js

```diff
@@ -139,11 +151,11 @@ const errorMap = {
 };
 
 class JsHintValidator extends Validator {
-  constructor(source, enabledLibraries) {
-    super(source, 'javascript', errorMap);
-
-    this._jshintOptions = defaults(clone(jshintrc), {predef: []});
-    enabledLibraries.forEach((libraryKey) => {
+  constructor(source, analyzer) {
+    super(source, 'javascript', errorMap, analyzer);
+    const jshintConfig = this._getConfig(analyzer);
+    this._jshintOptions = defaults(clone(jshintConfig), {predef: []});
+    analyzer.enabledLibraries.forEach((libraryKey) => {
       if (!(libraryKey in libraries)) {
         return;
       }
```

Comment:
To clarify, my proposal would be:

* Just define the one default `jshintrc`
* In the `_getConfig()` method/function, deep-copy that default object and then mutate the copy according to the project properties, meaning what’s there now plus the logic above.